### PR TITLE
ci: switch to vedantmgoyal2009/winget-releaser

### PR DESCRIPTION
**Pull request by lewisakura**
Works on Linux which should in theory be faster than a Windows runner, + it uses Komac which ironically has much better support for Winget manifest generation than Microsoft's own tooling.

### CODE REVIEW

File: .github/workflows/winget-submission.yml [hunk 1]
```diff
@@ -1,33 +1,17 @@
...
-          # Fetching latest release from GitHub
-          $github = Invoke-RestMethod -uri "https://api.github.com/repos/vencord/vesktop/releases"
-          $targetRelease = $github | Select-Object -First 1
-          $installerUrl = $targetRelease | Select-Object -ExpandProperty assets -First 1 | Where-Object -Property name -match 'Vesktop-Setup.*?exe' | Select-Object -ExpandProperty browser_download_url
-          $packageVersion = $targetRelease.tag_name.Trim("v")
-
-          # Update package using wingetcreate
-          Invoke-WebRequest https://aka.ms/wingetcreate/latest -OutFile wingetcreate.exe
-          .\wingetcreate.exe update $packageId --version $packageVersion --urls "$installerUrl" --submit --token $env:WINGET_PAT
+        uses: vedantmgoyal2009/winget-releaser@v2
```
Comment (Vendicated) [inline .github/workflows/winget-submission.yml:13]: never use tags for untrusted actions, always pin to long commit hash

Comment (Vendicated) [reply .github/workflows/winget-submission.yml]: especially in cases like this where you're passing tokens to it

**Comment by exhq** [comment]
:trollface:

**Comment by Vendicated** [comment]
is this mergeable? i have no idea

**Comment by mantikafasi** [comment]
Try and see

**Comment by lewisakura** [comment]
yeah it should be fine

**Pull request merged**